src/fprime/fbuild/interaction.py

```diff
@@ -69,41 +76,98 @@ def run_impl(deployment: Path, path: Path, platform: str, verbose: bool):
         lines = lines[11:]  # Remove old header
         with open(dest, "a") as file_handle:
             file_handle.write("".join(lines))
-    removals = [path / "mod.mk", path / "Makefile", Path(hpp_src), Path(cpp_src)]
+
+    removals = [Path(hpp_src), Path(cpp_src)]
     for removal in removals:
         os.remove(removal)
     return True
 
 
 def add_to_cmake(list_file: Path, comp_path: Path):
-    """Adds new component to CMakeLists.txt"""
+    """ Adds new component or port to CMakeLists.txt"""
+    print("[INFO] Found CMakeLists.txt at '{}'".format(list_file))
+    with open(list_file, "r") as f:
+        lines = f.readlines()
+        index = 0
+        while "add_fprime_subdirectory" not in lines[index]:
+            index += 1
+        while "add_fprime_subdirectory" in lines[index]:
+            index += 1
+
+    if not confirm(
+        "Add component {} to {} {}?".format(
+            comp_path, list_file, "at end of file before topology inclusion?"
+        )
+    ):
+        return False
+
+    addition = (
+        'add_fprime_subdirectory("${CMAKE_CURRENT_LIST_DIR}/' + str(comp_path) + '/")\n'
+    )
+    if addition in lines:
+        print("Component already added to CMakeLists.txt")
+        return True
+    lines.insert(index, addition)
+    with open(list_file, "w") as f:
+        f.write("".join(lines))
+    return True
+
+
+def regenerate(cmake_list_file):
+    # Ensures build directory exists, then refreshes build cache
+    build_dir = Path(cmake_list_file.parent, "build-fprime-automatic-native")
```

Comment:
We probably don't want to assume the build directory name. Can you use the same method the rest of fprime-util uses to determine the build directory name? I think it's by instantiating the build class.